obspy/clients/fdsn/mass_downloader/__init__.py

```diff
@@ -0,0 +1,503 @@
+results in
+
+``some_folder/BW/FURT/.BHZ.2014-10-27T16-37-23Z.2014-10-27T16-37-33Z.mseed``.
+
+The download helpers will create any non-existing folders along the path.
+
+**Option 3: Custom Function**
+
+The most complex but also most powerful possibility is to use a function which
+will be evaluated to determine the filename. **If the function returns**
+``True`` **, the MiniSEED file is assumed to already be available and will not
+be downloaded again; keep in mind that in that case no station data will be
+downloaded for that channel.** If it returns a string, the MiniSEED file will
+be saved to that path. Utilize closures to use any other parameters in the
+function. This hypothetical function checks if the file is already in a
+database and otherwise returns a string which will be interpreted as a
+filename.
+
+>>> def get_mseed_storage(network, station, location, channel, starttime,
+...                       endtime):
+...     # Returning True means that neither the data nor the StationXML file
+...     # will be downloaded.
+...     if is_in_db(network, station, location, channel, starttime, endtime):
+...         return True
+...     # If a string is returned the file will be saved in that location.
+...     return os.path.join(ROOT, "%s.%s.%s.%s.mseed." % (network, station,
+...                                                       location, channel))
+>>> mseed_storage = get_mseed_storage
+
+.. note::
+
+    No matter which approach is chosen, if a file already exists, it will not
+    be overwritten; it will be parsed and the download helper class will
+    attempt to download matching StationXML files.
+
+
+Storing StationXML files
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The same logic applies to the StationXML files. This time the rules are set by
+the ``stationxml_storage`` argument of the
+:func:`~obspy.clients.fdsn.mass_downloader.mass_downloader.MassDownloader.download`
+method of the
+:class:`~obspy.clients.fdsn.mass_downloader.mass_downloader.MassDownloader`
+class. StationXML files will be downloaded on a per-station basis thus all
+channels and locations from one station will end up in the same StationXML
+file.
+
+**Option 1: Folder Name**
+
+A simple string will be interpreted as a folder name. This example will save
+the files to ``"stations/NETWORK.STATION.xml"``, e.g. to
+``"stations/BW.FURT.xml"``.
+
+>>> stationxml_storage = "stations"
+
+**Option 2: String Template**
+
+Another option is to provide a string formatting template, e.g.
+
+>>> stationxml_storage = "some_folder/{network}/{station}.xml"
+
+will write to ``"some_folder/NETWORK/STATION.xml"``, in this case for example
+to ``"some_folder/BW/FURT.xml"``.
+
+.. note::
+
+    If the StationXML file already exists, it will be opened to see what is in
+    the file. In case it does not contain all necessary channels, it will be
+    deleted and **only those channels needed in the current run will be
+    downloaded again**. Pass a custom function to the ``stationxml_path``
```

Comment:
Only those for which the current downloading run did download some MiniSEED files. Is that not clear from the sentence?